### PR TITLE
Make v1beta1 be the only one in the codebase

### PR DESCRIPTION
**Pull request by arunmk**
Fixes VCDA-3626
This also needs a golang compiler version update. v1.17 is supported in k8s 1.23 onwards.
Also adds a `make vendor` that will get the vendored bits.

Please upgrade to golang v1.17 before using the branch

---
This change is [Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/111)

### CODE REVIEW

**Review by abaruni — APPROVED** [review]
> Reviewed 16 of 878 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/111)* status: 16 of 878 files reviewed, all discussions resolved (waiting on @Anirudh9794 and @sahithi)

**Review by Anirudh9794 — APPROVED** [review]
> Reviewed 55 of 878 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/111)* status: 55 of 878 files reviewed, all discussions resolved (waiting on @sahithi)

File: go.mod [hunk 1]
```diff
@@ -1,26 +1,97 @@
...
 	k8s.io/klog v1.0.0
-	k8s.io/klog/v2 v2.9.0
-	sigs.k8s.io/cluster-api v0.4.2
-	sigs.k8s.io/controller-runtime v0.10.3
+	k8s.io/klog/v2 v2.60.1
+	sigs.k8s.io/cluster-api v1.1.3
+	sigs.k8s.io/controller-runtime v0.11.2
+)
+
+require (
```
Comment (sahithi) [inline go.mod:28]: Is there any difference between this and the top require block?

**Review by sahithi — APPROVED** [review]
:lgtm:

File: go.mod (hunk 1 above)
Comment (arunmk) [reply go.mod:28]: This was added by the `go mod --compat=1.17`. I have not dug much into it.

**Pull request merged**